wcfsetup/install/files/lib/acp/form/LanguageEditForm.class.php

```diff
@@ -31,79 +25,12 @@ public function readParameters()
     {
         parent::readParameters();
 
-        if (isset($_REQUEST['id'])) {
-            $this->languageID = \intval($_REQUEST['id']);
-        }
-        $this->language = new Language($this->languageID);
-        if (!$this->language->languageID) {
-            throw new IllegalLinkException();
-        }
-    }
-
-    /**
-     * @inheritDoc
-     */
-    protected function validateLanguageCode()
-    {
-        if ($this->language->languageCode != \mb_strtolower($this->languageCode)) {
-            parent::validateLanguageCode();
+        if (isset($_REQUEST['id']) && \is_numeric($_REQUEST['id'])) {
```

Comment:
Not technically new code, but we don't use `$_REQUEST` in new code. As the `id` should always be part of the query string, not the body, simply using `$_GET` should be sufficient. In this case please perform the validation like this:

```
        $parameters = Helper::mapQueryParameters(
            $_GET,
            <<<'EOT'
                array {
                    id?: positive-int
                }
                EOT
        );

        // $parameters['id'] is now either a positive int or does not exist, no need for "numeric" checks or manual casting.
```

Reply:
Do you want the `Helper` to read the ID? That seems a little bit overkill for preventing type-errors.

Reply:
Yes, please use Valinor using `Helper::mapQueryParameters`. Valinor is the structual validator of choice beginning with WoltLab Suite 6.0. Consistency in validation is important, performance is not super critical in the ACP (and neither for the vast majority of frontend actions) and it's certainly much cleaner to use that than to repeat `$_REQUEST['id']` a total of 3 times.